src/check/parse/IR.zig

```diff
@@ -1517,6 +1607,29 @@ pub const NodeStore = struct {
             patterns: []const PatternIdx,
             region: Region,
         },
+
+        pub fn toSExpr(self: @This(), env: *base.ModuleEnv, ir: *IR) anyerror!sexpr.Node {
+            switch (self) {
+                .ident => |ident| {
+                    const token = ir.tokens.tokens.get(ident.ident_tok);
+
+                    // TODO why does using this text value cause a segfault?
+                    // it must be a lifetime issue...
+                    const text = env.idents.getText(token.extra.interned);
+                    _ = text;
+
+                    return .{
+                        .node = .{
+                            .value = "ident",
+                            .children = &.{
+                                .{ .string = "TODO" },
+                            },
+                        },
+                    };
```

Comment:
@bhansconnect -- I must be doing something silly here. This is causing a segfault if I use the text value. 

The text should be an interned string in the env which I expect to have a lifetime at least as long as the SExpr IR. 

I've tried also using the arena to allocate and duplicate the bytes across but I'm missing something here. 

Any ideas?

Reply:
You need to allocate the children. `&.{ ... }` is a stack allocated array an will be garbage after returning.

```zig
                    var children = try env.arena.allocator().alloc(sexpr.Node, 1);
                    children[0] = .{ .string = text };

                    return .{
                        .node = .{
                            .value = "ident",
                            .children = children,
                        },
                    };
```

Reply:
Fixed, thank you very much 🙏